auth-jwt/auth-jwt-api/src/main/java/com/duckduckgo/authjwt/api/AuthJwtValidator.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.authjwt.api
+
+import java.time.Instant
+
+/**
+ * This class provides methods for checking the validity of JWTs obtained from Auth API V2 and parsing claims included in those tokens.
+ */
+interface AuthJwtValidator {
```

Comment:
@lmac012 curious question, why separate this module from subscription-impl? As much as I see the APIs are very well defined, I don’t see the need of separating it from subscription-impl.

Reply:
The main reason is to encapsulate the third-party library used for validating JWTs. This approach ensures that types defined by the third-party library aren’t used throughout the `subscriptions-impl` module. If we ever need to migrate to a different library, it won't require a lot of refactoring. Putting this code in its own module is a suggestion I received in a TD conversation, and while I don’t feel strongly about it, I do see some value in this approach.